R/03b-plot-catchments.R

```diff
@@ -0,0 +1,499 @@
+    shape = 23,
+    fill = 'yellow',
+    color = 'black',
+    size = 2,
+    stroke = 1
+  ) +
+  geom_sf_text(
+    data = csd_centroids_nudged,
+    aes(label = csd_name), 
+    size = 4, 
+    fontface = 'bold'
+  ) +
+  # Colors for catchment areas
+  scale_fill_viridis_d(
+    option = "turbo",
+    name = 'Service BC Facility',
+    na.value = "darkgray"
+  ) +
+  # Alpha for assignment method
+  scale_alpha_manual(
+    values = c("drive_time" = 0.9, "nearest_facility" = 0.4),
+    name = "Assignment Method",
+    labels = c("Drive Time", "Direct Assignment")
+  ) +
+  coord_sf(expand = FALSE) + # remove whitespace
+  labs(
+    title = 'Service BC Facility Complete Catchment Areas',
+    subtitle = 'All areas assigned to nearest facility\nLighter areas show new assignments based on proximity to facilities'
+  ) +
+  MAP_THEME +
+  theme(legend.position = "none")  # Remove legends
+
+complete_map
+
+# Save the complete catchments map
+ggsave(
+  filename = glue("pilot_complete_catchments.png"),
+  path = maps_folder,
+  plot = complete_map,
+  width = 12,
+  height = 10,
+  dpi = 300
+)
+
+# ----------------------------------------------------------------------------
+# Map 4: Provincial view of all catchments (without unassigned areas)
+# ----------------------------------------------------------------------------
+
+# Create original catchment areas (with unassigned areas)
+province_original_catchments <- db_shapefile |>
+  left_join(complete_assignments, by = "dbid") |>
+  filter(assignment_method == 'drive_time') |>  
+  # Group by facility and merge geometries
+  group_by(assigned) |>
+  summarize(
+    n_dbs = n(),
+    assignment_method = "drive_time",
+    .groups = "drop"
+  ) |>
+  # Simplify the geometries for better mapping performance
+  ms_simplify(keep = 0.01, keep_shapes = TRUE)
+
+# Create "unassigned" areas
+unassigned_dbs <- db_shapefile |>
+  left_join(complete_assignments, by = "dbid") |>
+  filter(assignment_method == 'nearest_facility') |>
+  # Group and merge geometries
+  summarize(
+    n_dbs = n(),
+    assigned = "Unassigned", # Label for these areas
+    assignment_method = "none",
+    .groups = "drop"
+  ) |>
+  # Simplify the geometries
+  rmapshaper::ms_simplify(keep = 0.01, keep_shapes = TRUE)
+
+# Build map with original catchments
+province_original_map <- ggplot() + 
+  # Base BC map
+  geom_sf(
+    data = bc_map, 
+    fill = 'white', 
+    color = 'black'
+  ) +
+  # Unassigned areas
+  geom_sf(
+    data = unassigned_dbs,
+    fill = 'darkgray',
+    color = 'darkgrey',
+    alpha = 0.5
+  ) +
+  # Facility catchment areas
+  geom_sf(
+    data = province_original_catchments, 
+    aes(fill = assigned),
+    color = 'darkgrey',
+    alpha = 0.7
+  ) +
+  # Service BC locations
+  geom_sf(
+    data = sbc_locs,
+    shape = 23,
+    fill = 'yellow',
+    color = 'black',
+    size = 2,
+    stroke = 1
+  ) +
+  # Colors for catchment areas
+  scale_fill_viridis_d(
+    option = "turbo",
```

Comment:
Are you attached to _turbo_ or do you mind if we use the _mako_ (or _vridis_) option?  SBC locations pop with _mako_ in particular, but they are both less aggressive schemes and so easier to look at.

Reply:
I am not attached! Will update to mako :)

Reply:
Actually after looking at the mako map, I find it harder to distinguish between the different 'catchments', especially on the full province map. I Like the overall look of it better (with the yellow popping better you're right), but it (imo) does blend the catchment colors together a little bit too much.